src/Disks/ObjectStorages/MetadataStorageFromPlainRewritableObjectStorage.cpp

```diff
@@ -82,7 +88,25 @@ void MetadataStorageFromPlainRewritableObjectStorage::load()
 
     std::set<std::string> set_of_remote_paths;
 
-    if (!object_storage->existsOrHasAnyChild(metadata_key_prefix))
+    bool has_metadata = object_storage->existsOrHasAnyChild(metadata_key_prefix);
+
+    if (is_initial_load)
+    {
+        /// Use iteration to determine if the disk contains data.
+        /// LocalObjectStorage creates an empty top-level directory even when no data is stored,
+        /// unlike blob storage, which has no concept of directories, therefore existsOrHasAnyChild
+        /// is not applicable.
+        auto common_key_prefix = fs::path(object_storage->getCommonKeyPrefix()) / "";
+        bool has_data = object_storage->iterate(common_key_prefix, 0)->isValid();
```

Comment:
Does it make sense to do like this? (`isRemote` is an already existing method)
```suggestion
        bool has_data = object_storage->isRemote() ? object_storage->existsOrHasAnyChild(common_key_prefix) : object_storage->iterate(common_key_prefix, 0)->isValid();
```